sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareListFilesAndDirectoriesOptions.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * @param prefix the prefix.
+     * @return updated options.
+     */
+    public ShareListFilesAndDirectoriesOptions setPrefix(String prefix) {
+        this.prefix = prefix;
+        return this;
+    }
+
+    /**
+     * @return max results per page for this listing operation.
+     */
+    public Integer getMaxResultsPerPage() {
+        return maxResultsPerPage;
+    }
+
+    /**
+     * Sets the max results per page for a listing operation.
+     *
+     * @param maxResultsPerPage the max results per page.
+     * @return updated options.
+     */
+    public ShareListFilesAndDirectoriesOptions setMaxResultsPerPage(Integer maxResultsPerPage) {
+        this.maxResultsPerPage = maxResultsPerPage;
+        return this;
+    }
+
+    /**
+     * Note that setting timestamps, etag, attributes, or permission key will also set this option as true. Attempting
+     * to set it back to false while any of these options are true will be unsuccessful.
+     *
+     * Including extended info in a listing operation can result in a more expensive operation, but will present
+     * more accurate information on the listing item.
+     *
+     * @return whether to include extended info on this listing operation.
+     */
+    public Boolean includeExtendedInfo() {
+        return includeExtendedInfo;
+    }
+
+    /**
+     * Note that setting timestamps, etag, attributes, or permission key will also set this option as true. Attempting
+     * to set it back to false will be unsuccessful.
+     *
+     * Sets the prefix for a listing operation.
+     *
+     * Including extended info in a listing operation can result in a more expensive operation, but will present
+     * more accurate information on the listing item.
+     *
+     * @param includeExtendedInfo whether to include extended info.
+     * @return updated options.
+     * @throws IllegalStateException Throws when attempting to set null when other parameters require it to be true.
+     */
+    public ShareListFilesAndDirectoriesOptions setIncludeExtendedInfo(Boolean includeExtendedInfo) {
+        if (includeTimestamps || includeETag || includeAttributes || includePermissionKey) {
+            throw logger.logExceptionAsError(
+                new IllegalStateException("includeExtendedInfo must be true in the current state."));
+        }
+        this.includeExtendedInfo = includeExtendedInfo;
+        return this;
+    }
+
+    /**
+     * @return whether to include timestamps on this listing operation.
+     */
+    public boolean includeTimestamps() {
+        return includeTimestamps;
+    }
+
+    /**
+     * @param includeTimestamps whether to include timestamps on this listing operation.
+     * @return updated options
+     */
+    public ShareListFilesAndDirectoriesOptions setIncludeTimestamps(boolean includeTimestamps) {
+        this.includeTimestamps = includeTimestamps;
+        if (includeTimestamps) {
+            this.includeExtendedInfo = true;
+        }
```

Comment:
This isn't necessary as well. service will do the right thing. Please remove this and similar logic.